api/src/app/modules/attachment/attachment.service.ts

```diff
@@ -110,8 +111,8 @@ export class AttachmentService extends DataService<Attachment, Repository<Attach
   }
 
   async isViewAuthorized(entity: Attachment, user?: User): Promise<boolean> {
-
-    if (entity.attachmentTypeCode == AttachmentTypeEnum.PUBLIC_NOTICE || entity.attachmentTypeCode == AttachmentTypeEnum.SUPPORTING_DOC) {
+    const attachmentTypeCode = entity.attachmentType?.code;
+    if (attachmentTypeCode == AttachmentTypeEnum.PUBLIC_NOTICE || attachmentTypeCode == AttachmentTypeEnum.SUPPORTING_DOC) {
       // These document types are viewable by the public.
```

Comment:
I don't remember if supporting documents should be visible to the public. Interactions should definitely NOT be.

Reply:
The pre-existing code had this same logic (public can see supporting documents) so lets keep it that way.

Reply:
Yes, the business logic for condition did not get changed. What I was trying to do is to use the same constant for value of 'attachmentTypeCode'. However, I do change it to get value from the AttachmentTypeCode relation entity that's available from Attachment entity since the other `@RelationId` is for some reason empty that is causing the checking fails. I do document this point in ticket.